fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java

```diff
@@ -121,30 +121,18 @@ private static void analyzeBuiltinAggFunction(FunctionCallExpr functionCallExpr)
         }
 
         if (fnName.getFunction().equals(FunctionSet.GROUP_CONCAT)) {
-            if (functionCallExpr.getChildren().size() > 2 || functionCallExpr.getChildren().isEmpty()) {
+            if (functionCallExpr.getChildren().size() - fnParams.getOrderByElemNum() < 2) {
                 throw new SemanticException(
-                        "group_concat requires one or two parameters: " + functionCallExpr.toSql(),
+                        "group_concat requires at least one parameter: " + functionCallExpr.toSql(),
                         functionCallExpr.getPos());
             }
 
-            if (fnParams.isDistinct()) {
-                throw new SemanticException("group_concat does not support DISTINCT", functionCallExpr.getPos());
-            }
-
-            Expr arg0 = functionCallExpr.getChild(0);
-            if (!arg0.getType().isStringType() && !arg0.getType().isNull()) {
+            int sepPos = functionCallExpr.getParams().exprs().size() - functionCallExpr.getParams().getOrderByElemNum() - 1;
+            Expr arg1 = functionCallExpr.getChild(sepPos);
```

Comment:
Can the separator be a expr like `upper('-')`?

Reply:
yes
```
mysql> select group_concat(name SEPARATOR upper('-')) from ss;
+--------------------------------------------------------------------------------------------------------------------------------+
| group_concat(name SEPARATOR upper('-'))                                                                                        |
+--------------------------------------------------------------------------------------------------------------------------------+
| May-Ti-欧阳诸葛方程-Tom-Tom-张三此地无银三百两-张三掩耳盗铃-Ti-李四大闹天空-Tom-Tom-王武程咬金                                 |
+--------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.45 sec)
```